File: OCH_Tao/ABC/B/255.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input.shift().split(" ")[0], 10);
const A = new Set(input.shift().split(" ").map(x => parseInt(x, 10)));
let L = [];
let R = [];
for (let i = 0; i < N; i++) {
  if (A.has(i + 1)) {
    R.push(input[i].split(" ").map(x => parseInt(x, 10)));
  } else {
    L.push(input[i].split(" ").map(x => parseInt(x, 10)));
  }
}
let ans = new Array(L.length).fill(Infinity);
for (const i of R) {
  for (let j = 0; j < L.length; j++) {
    ans[j] = Math.min(ans[j], ((i[0] - L[j][0]) ** 2 + (i[1] - L[j][1]) ** 2) ** (1 / 2));
  }
}
console.log(Math.max(...ans));

File: OCH_Tao/ABC/B/255.py
import numpy as np
N,K = map(int,input().split())
A = set(list(map(int,input().split())))
R = []
L = []
for i in range(1,N+1):
  if i in A:
    R.append(np.array(list(map(int,input().split()))))
  else:
    L.append(np.array(list(map(int,input().split()))))
ans = [np.inf]*len(L)
for i in R:
  for j in range(len(L)):
    ans[j]=min(ans[j],np.linalg.norm(i-L[j]))
print(max(ans))